app/qan/reporter.go

```diff
@@ -89,10 +89,21 @@ func SparklineData(qr *Reporter, endTs int64, intervalTs int64, queryClassId uin
 		queryLogArrRaw[(ql.Start_ts).Unix()] = ql
 	}
 
+	intervalTime := end.Sub(begin).Minutes()
+	amountOfPoints := int64(maxAmountOfPoints)
+	if intervalTime < maxAmountOfPoints {
+		amountOfPoints = int64(intervalTime)
+	}
 	var i int64
 	for i = 0; i < amountOfPoints; i++ {
 		ts := endTs - i*intervalTs
 		val, ok := queryLogArrRaw[ts]
+
+		// skip first or last point if they are empty
+		if (i == 0 || i == amountOfPoints-1) && !ok {
```

Comment:
Why they are empty?

Reply:
because of SQLQuery windows calculation. or according last point - still ton received data from agent - for small time-ranges